games/game_stalkeranomaly.py

```diff
@@ -132,20 +141,724 @@ def getContentsFor(self, tree: mobase.IFileTree) -> List[int]:
         return self.content
 
 
+class IVec3:
+    def __init__(self, x, y, z):
+        self.x = x
+        self.y = y
+        self.z = z
+
+    def __str__(self) -> str:
+        return f"{self.x}, {self.y}, {self.z}"
+
+    def set(self, x, y, z):
+        self.x = x
+        self.y = y
+        self.z = z
+
+
+class IVec4(IVec3):
+    def __init__(self, x, y, z, w):
+        super().__init__(x, y, z)
+        self.w = w
+
+    def __str__(self) -> str:
+        return f"{self.x}, {self.y}, {self.z}, f{self.w}"
+
+    def set(self, x, y, z, w):
+        super().set(x, y, z)
+        self.w = w
+
+
+class IFlag:
+    def __init__(self, flag: int):
+        self._flag = flag
+
+    def __str__(self) -> str:
+        return str(self._flag)
+
+    def assign(self, mask):
+        self._flag = mask
+
+    def has(self, mask) -> bool:
+        return bool((self._flag & mask) == mask)
+
+    def set(self, mask):
+        self._flag |= mask
+
+    def remove(self, mask):
+        self._flag &= ~mask
+
+
+class IReader:
```

Comment:
I get the idea behind this but most reading "buffer" features (`_read_`, `read`, `tell`, `eof`, ...), can be found in `io.BytesIO` so maybe it'd be better to extends `io.BytesIO` and only add the required methods?

Reply:
I'll try it, but it doesn't make a significant difference and I still need to implement nearly identical (but not quite) variations. This also adds additional methods that I don't really need right now.

Reply:
Never mind, I can't even access the buffer directly, I have to use a method to get it. I'm just going to stick with my own basic implementation

Reply:
Why would you need access to the underlying buffer?

Reply:
For __len__ and peek. Seems silly, but my other points still hold true.

Reply:
You can compute the length when creating the object since it's constant (or `len(self.getbuffer())` and you never use `peek`. You only need to implement the function that read specific value.

On another topic, if you have a lots of save and wants to improve performance, you're probably better reading multiple things at a time with `struct`.

Reply:
I'll worry about performance another time, it's fast enough and I was more inclined to mimic the games own reader and serialization of classes.